lib/sycamore/sycamore/evaluation/evaluate.py

```diff
@@ -0,0 +1,139 @@
+            document = EvaluationDataPoint()
+            datapoint["custom_question_augmentation"] = custom_question_augmentation
+            datapoint["question_augmentation_filter"] = question_augmentation_filter
+            document.raw = datapoint
+            document.ground_truth_answer = datapoint["Answer"]
+            document.filters = datapoint.get("Filters", {})
+            result += [{"doc": document.serialize()}]
+        return result
+
+    def run_evaluation(self, ctx: Context, index: str, **kwargs):
+        custom_question_augmentation = str(self.custom_question_augmentation)
+        question_augmentation_filter = str(self.question_augmentation_filter)
+        input_docset = ctx.read.json(
+            paths=self.gt_path,
+            doc_extractor=lambda json_dict: QualityAssessment.create_evaluation_datapoint(
+                json_dict, custom_question_augmentation, question_augmentation_filter
+            ),
+        ).map(add_filters_to_question)
+
+        pipeline = EvaluationPipeline(
+            index=index, os_client_args=self.os_client_args, os_config=self.rag_config, metrics=self.metrics
+        )
+        query_level_metrics, aggregated_metrics = pipeline.execute(input_docset)
+        return query_level_metrics.take_all(), aggregated_metrics
+
+
+class Evaluate:
+    """
+    The Evaluate runs the evaluation test on
+    Index or list of indices against a ground truth
+
+    Args:
+        context: The Sycamore context to use
+        index: Index or list of Index
+        assessment: The Assessment to run
+        GT_path: The path to ground truth
+        rag_config: Configration for RAG
+        os_client_args: Configration for connecting to opensearch
+        custom_question_augmentation: Custom String for Augmenting Question
+        question_augmentation_filter: Filters values to be use in custom Question Augmentation
+
+    Returns:
+        Two EvaluationDataPoint, one for query level information and another with aggregate information
+
+    Example:
+        context = sycamore.init()
+
+        custom_question_augmentation = "{}, The product code is {}."
+        question_augmentation_filter = 'properties._product_codes'
+
+        assessment = QualityAssessment(os_client_args=OS_CLIENT_ARGS,
+            rag_config= OS_CONFIG,
+            GT_path = './test.json',
+            custom_question_augmentation=custom_question_augmentation,
+            question_augmentation_filter = question_augmentation_filter)
+        evaluate = Evaluate(context,'index_V1',assessment)
+    """
+
+    def __init__(self, context: Context, index: Union[str, List[str]], assessment: Assessment, **kwargs):
+        super().__init__()
+
+        if isinstance(index, str):
+            self.result = {index: assessment(context, index)}
+        elif isinstance(index, List) and all(isinstance(i, str) for i in index):
+            self.result = {idx: assessment(context, idx) for idx in index}
```

Comment:
I would recommend using a `@property` to access `result` since presumably you don't want client code modifying it directly.

Reply:
Thanks for catching this, using property for saving result.